From abaf0e1db0ac7f552082e0a44dc9b4866e3151d1 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Mon, 8 Apr 2024 18:31:44 +0200
Subject: [PATCH 0762/1695] dts: stm32: notify PMIC PKEY IRQ on stm32mp25
 boards

Notify PMIC power key interrupts to the non secure world.
Those interrupts are generated by the wake-up button.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9bf64de140bb6b0a746fb12d67085ce02adb93e3 dts: stm32: notify PMIC PKEY IRQ on stm32mp25 boards]
Change-Id: I68e919fb189bf1a8857cbb22d64d69991b0ba515
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373300
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts  | 3 +++
 core/arch/arm/dts/stm32mp257f-ev1.dts | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 6f77d4e49..33ca16a60 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include <dt-bindings/mfd/st,stpmic2.h>
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/tamper/st,stm32mp25-tamp.h>
@@ -75,6 +76,8 @@
 		compatible = "st,stpmic2";
 		reg = <0x33>;
 		st,wakeup-pin-number = <1>;
+		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
+		st,notif-it-id = <0 1>;
 		wakeup-parent = <&pwr>;
 		status = "okay";
 
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 4f0bef74c..a541c868c 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include <dt-bindings/mfd/st,stpmic2.h>
 #include <dt-bindings/regulator/st,stm32mp25-regulator.h>
 #include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/soc/stm32mp25-rif.h>
@@ -75,6 +76,8 @@
 		compatible = "st,stpmic2";
 		reg = <0x33>;
 		st,wakeup-pin-number = <1>;
+		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
+		st,notif-it-id = <0 1>;
 		wakeup-parent = <&pwr>;
 		status = "okay";
 
-- 
2.47.2

